src/utilities/introspectionQuery.js

```diff
@@ -124,95 +124,117 @@ export type IntrospectionType =
+  | IntrospectionInterfaceType
+  | IntrospectionUnionType
+  | IntrospectionEnumType;
+
+export type IntrospectionInputType =
+  | IntrospectionScalarType
+  | IntrospectionEnumType
+  | IntrospectionInputObjectType;
+
+export type IntrospectionScalarType = {|
+  +kind: 'SCALAR';
+  +name: string;
+  +description: ?string;
+|};
+
+export type IntrospectionObjectType = {|
+  +kind: 'OBJECT';
+  +name: string;
+  +description: ?string;
+  +fields: Array<IntrospectionField>;
+  +interfaces: Array<IntrospectionNamedTypeRef<IntrospectionInterfaceType>>;
+|};
+
+export type IntrospectionInterfaceType = {|
+  +kind: 'INTERFACE';
+  +name: string;
+  +description: ?string;
+  +fields: Array<IntrospectionField>;
+  +possibleTypes: Array<IntrospectionNamedTypeRef<IntrospectionObjectType>>;
+|};
+
+export type IntrospectionUnionType = {|
+  +kind: 'UNION';
+  +name: string;
+  +description: ?string;
+  +possibleTypes: Array<IntrospectionNamedTypeRef<IntrospectionObjectType>>;
+|};
+
+export type IntrospectionEnumType = {|
+  +kind: 'ENUM';
+  +name: string;
+  +description: ?string;
+  +enumValues: Array<IntrospectionEnumValue>;
+|};
+
+export type IntrospectionInputObjectType = {|
+  +kind: 'INPUT_OBJECT';
+  +name: string;
+  +description: ?string;
+  +inputFields: Array<IntrospectionInputValue>;
+|};
+
+export type IntrospectionListTypeRef<T> = {|
+  +kind: 'LIST',
+  +ofType?: T,
+|};
+
+export type IntrospectionNonNullTypeRef<T> = {|
+  +kind: 'NON_NULL',
+  +ofType?: T,
+|};
 
 export type IntrospectionTypeRef =
-  | IntrospectionNamedTypeRef
-  | IntrospectionListTypeRef
-  | IntrospectionNonNullTypeRef;
-
-export type IntrospectionNamedTypeRef = {
-  kind: string;
-  name: string;
-};
-
-export type IntrospectionListTypeRef = {
-  kind: 'LIST';
-  ofType?: IntrospectionTypeRef;
-};
-
-export type IntrospectionNonNullTypeRef = {
-  kind: 'NON_NULL';
-  ofType?: IntrospectionTypeRef;
-};
-
-export type IntrospectionField = {
-  name: string;
-  description: ?string;
-  args: Array<IntrospectionInputValue>;
-  type: IntrospectionTypeRef;
-  isDeprecated: boolean;
-  deprecationReason: ?string;
-};
-
-export type IntrospectionInputValue = {
-  name: string;
-  description: ?string;
-  type: IntrospectionTypeRef;
-  defaultValue: ?string;
-};
-
-export type IntrospectionEnumValue = {
-  name: string;
-  description: ?string;
-  isDeprecated: boolean;
-  deprecationReason: ?string;
-};
-
-export type IntrospectionDirective = {
-  name: string;
-  description: ?string;
-  locations: Array<DirectiveLocationEnum>;
-  args: Array<IntrospectionInputValue>;
-};
+  | IntrospectionNamedTypeRef<IntrospectionType>
+  | IntrospectionListTypeRef<IntrospectionTypeRef>
+  | IntrospectionNonNullTypeRef<IntrospectionTypeRef>;
+
+export type IntrospectionOutputTypeRef =
+  | IntrospectionNamedTypeRef<IntrospectionOutputType>
+  | IntrospectionListTypeRef<IntrospectionOutputTypeRef>
+  | IntrospectionNonNullTypeRef<IntrospectionOutputTypeRef>;
```

Comment:
This is technically not correct since a non-null cannot wrap a non-null. Elsewhere where we encode types that account for list and non-null we account for this. It might look something like:

```js
export type IntrospectionOutputTypeRef =
  | IntrospectionNamedTypeRef<IntrospectionOutputType>
  | IntrospectionListTypeRef<IntrospectionOutputTypeRef>
  | IntrospectionNonNullTypeRef<
    | IntrospectionNamedTypeRef<IntrospectionOutputType>
    | IntrospectionListTypeRef<IntrospectionOutputTypeRef>
  >;
```

Similar below